sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -355,14 +354,12 @@ public <T> T findById(String containerName, Object id, Class<T> domainType) {
         CosmosEntityInformation<?, ?> cosmosEntityInformation = CosmosEntityInformation.getInstance(domainType);
         String containerPartitionKey = cosmosEntityInformation.getPartitionKeyFieldName();
         if ("id".equals(containerPartitionKey) && id != null) {
-            try {
-                return findById(id, domainType, new PartitionKey(CosmosUtils.getStringIDValue(id)));
-            } catch (CosmosAccessException e) {
-                LOGGER.warn(e.getMessage());
-                return null;
-            }
+            return findById(id, domainType, new PartitionKey(id));
+        }
+        if (!this.pointReadWarningLogged) {
```

Comment:
NIT: This is not thread-safe - you might end-up with multiple logs. Not too critical (maybe even intentional because while possibly logging multiple times the check whether logging happened becomes cheaper) - so I leave it to you whether you decide to make it thread-safe or not.

If you do - change the pointReadWarningLogged to an AtomicBoolean

Then use 
if (this.pointReadWarningLogged.compareAndSet(false, true) {
    ...
}

Reply:
Discussed already with Kushagra, view was that it’s ok (without even thinking about the cost of thread-safe check). So I think it’s ok as is. Thanks 😊